Api reference min version docs (#2424)
canonical/ubuntu-advantage-client: docs/references/api.md

}
```

## Version Dependencies
The package name of Ubuntu Pro Client is `ubuntu-advantage-tools`.

> The documentation for each endpoint below includes the first version that includes that endpoint.

Reviewer: ```suggestion
The documentation for each endpoint below states the first version to include that endpoint.
```

Reviewer: alternatively: "The documentation for each endpoint below includes the first version to introduce that endpoint."

> If you depend on these endpoints, we recommend using a standard dependency version requirement to ensure that a sufficiently up to date version of the pro client is installed before trying to use it.

Reviewer: ```suggestion
If you depend on these endpoints, we recommend using a standard dependency version requirement to ensure that a sufficiently up-to-date version of the Pro Client is installed before trying to use it.
```

}
```

> ## Version Dependencies

Reviewer: (just in case I miss any, headings should ideally be in sentence case)

The package name of Ubuntu Pro Client is `ubuntu-advantage-tools`.

The documentation for each endpoint below includes the first version that includes that endpoint.

If you depend on these endpoints, we recommend using a standard dependency version requirement to ensure that a sufficiently up to date version of the pro client is installed before trying to use it.

```{important}
The `~` at the end of the version is important to ensure that dpkg version comparison works as expected.
```

For example, if your software is packaged as a deb, then you can add the following to your `Depends` list in your `control` file to ensure the installed version is at least 27.11~:
```
ubuntu-advantage-tools (>= 27.11~)
```

> ### Runtime Version Detection

Reviewer: ```suggestion
### Runtime version detection
```

> If you need to detect the current version at runtime, the most reliable way is to query dpkg.

Reviewer: ```suggestion
If you need to detect the current version at runtime, the most reliable way is to query `dpkg`.
```

```
dpkg-query --showformat='${Version}' --show ubuntu-advantage-tools
```

> If you need to compare versions at runtime, make sure you use the dpkg compare version algorithm.

Reviewer: ```suggestion
If you need to compare versions at runtime, make sure you use the `dpkg` compare version algorithm.
```

> For example, the following will exit 0 if the currently installed version of Pro client is at least 27.11~:

Reviewer: ```suggestion
For example, the following will exit 0 if the currently installed version of Pro Client is at least 27.11~:
```

```
dpkg --compare-versions "$(dpkg-query --showformat='${Version}' --show ubuntu-advantage-tools)" ge "27.11~"
```

> ### Runtime Feature Detection

Reviewer: ```suggestion
### Runtime feature detection
```

> As an alternative to version detection, you can use feature detection. This is easier to do when importing the api in python than it is when using the `pro api` subcommand.

Reviewer: ```suggestion
As an alternative to version detection, you can use feature detection. This is easier to do when importing the API in python than it is when using the `pro api` subcommand.
```
